### PR TITLE
docs: add examples for --format

### LINKED ISSUE #1422: Add examples for the flag "--format"
### What happened in your environment?

After upgraded to ORAS v1.2.0, I noticed that there are no any examples of the flag `--format` in ORAS CLI command help reference. Users might be hard to understand how to use `--format` in ORAS CLI commands

image

### What did you expect to happen?

It would be much user-friendly if we could add basic examples for the flag "--format" in the help reference. 

### How can we reproduce it?

Run "--help" in each oras command

### What is the version of your ORAS CLI?

oras v1.2.0

### What is your OS environment?

Ubuntu 22.06

### Are you willing to submit PRs to fix it?

- [ ] Yes, I am willing to fix it.

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1422 

**Please check the following list**:
- [ ]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

File: cmd/oras/root/attach.go [hunk 1]
```diff
@@ -77,6 +77,12 @@ Example - Attach an artifact with manifest annotations:
 Example - Attach file 'hi.txt' and add manifest annotations:
   oras attach --artifact-type doc/example --annotation "key=val" localhost:5000/hello:v1 hi.txt
 
+Example - Attach file 'hi.txt' and format output with JSON:
+  oras attach --artifact-type doc/example localhost:5000/hello:v1 hi.txt --format json
+
+Example - Attach file 'hi.txt' and format output with Go template:
+  oras attach --artifact-type doc/example localhost:5000/hello:v1 hi.txt --format go-template --template {{.digest}}
```
Comment (shizhMSFT) [inline cmd/oras/root/attach.go:84]: It is better to quote the template.
```suggestion
  oras attach --artifact-type doc/example localhost:5000/hello:v1 hi.txt --format go-template --template "{{.digest}}"
```

Comment (shizhMSFT) [reply cmd/oras/root/attach.go:84]: Same comment for other commands

File: cmd/oras/root/discover.go [hunk 2]
```diff
@@ -51,27 +51,30 @@ func discoverCmd() *cobra.Command {
...
-  oras discover -v -o tree localhost:5000/hello:v1
+  oras discover -v localhost:5000/hello:v1
 
 Example - Discover referrers with type 'test-artifact' of manifest 'hello:v1' in registry 'localhost:5000':
   oras discover --artifact-type test-artifact localhost:5000/hello:v1
 
 Example - Discover referrers of the manifest tagged 'v1' in an OCI image layout folder 'layout-dir':
   oras discover --oci-layout layout-dir:v1
-  oras discover --oci-layout -v -o tree layout-dir:v1
+  oras discover --oci-layout -v layout-dir:v1
```
Comment (shizhMSFT) [inline cmd/oras/root/discover.go:77]: Do we still need this line?

File: cmd/oras/root/manifest/fetch.go [hunk 3]
```diff
@@ -58,6 +58,9 @@ Example - Fetch the descriptor of a manifest from a registry:
 Example - Fetch the manifest digest from a registry similar to the resolve command:
   oras manifest fetch --format go-template --template '{{ .digest }}' localhost:5000/hello:v1
 
+Example - Fetch manifest and format output with JSON:
```
Comment (shizhMSFT) [inline cmd/oras/root/manifest/fetch.go:61]: manifest itself is JSON.
```suggestion
Example - Fetch manifest and output metadata encoded in JSON:
```

File: cmd/oras/root/pull.go [hunk 4]
```diff
@@ -83,6 +83,12 @@ Example - Pull files from a registry with certain platform:
 Example - Pull all files with concurrency level tuned:
   oras pull --concurrency 6 localhost:5000/hello:v1
 
+Example - Pull files and format output with JSON:
```
Comment (shizhMSFT) [inline cmd/oras/root/pull.go:86]: in JSON?

Comment (shizhMSFT) [reply cmd/oras/root/pull.go:86]: Same comment for other commands

File: cmd/oras/root/discover.go (hunk 2 above)
Comment (wangxiaoxuan273) [reply cmd/oras/root/discover.go]: removed this line

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**